src/inner_product_arg/mod.rs

```diff
@@ -0,0 +1,1079 @@
+use crate::{BTreeMap, BTreeSet, ToString, Vec};
+use crate::{BatchLCProof, Error, Evaluations, QuerySet};
+use crate::{LabeledCommitment, LabeledPolynomial, LinearCombination};
+use crate::{PCCommitterKey, PCVerifierKey};
+use crate::{PCRandomness, PCUniversalParams, Polynomial, PolynomialCommitment};
+
+use algebra_core::{
+    AffineCurve, Field, One, PrimeField, ProjectiveCurve, ToBytes, UniformRand, VariableBaseMSM,
+    Zero,
+};
+use core::{convert::TryInto, marker::PhantomData};
+use rand_core::RngCore;
+
+mod data_structures;
+pub use data_structures::*;
+
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+
+use digest::Digest;
+use std::iter::Iterator;
+
+/// Polynomial commitment scheme based on the discrete log problem.
+/// The specific construction is detailed in
+/// [[BCMS20, "Proof-Carrying Data from Accumulation Schemes"]][pcdas].
+/// If degree bounds are used, this implementation requires that points
+/// are sufficiently random or previously unknown such that the opening party cannot
+/// construct a commitment specialized to the point to replace shifted commitments.
+///
+/// [pcdas]: https://eprint.iacr.org/2020/499
```

Comment:
```suggestion

/// A polynomial commitment scheme based on the hardness of the 
/// discrete logarithm problem in prime-order groups.
/// The construction is detailed in
/// [[BCMS20]][pcdas].
///
/// To enforce degree bounds, it must be the case that the points at 
/// which a committed polynomial is evaluated are from a distribution that is
/// random conditioned on the polynomial. This is because degree bound
/// enforcement relies on checking a polynomial identity at this point. 
///
/// [pcdas]: https://eprint.iacr.org/2020/499
```